application/forms/PreferenceForm.php

```diff
@@ -202,6 +203,35 @@ public function createElements(array $formData)
             }
         }
 
+        $regEx = '/@media\s*\(\s*min-height\s*:\s*@prefer-(light|dark)-color-scheme\s*\)\s*,\s*\(\s*prefers-color-scheme\s*:\s*(light|dark)\s*\)\s*and\s*\(\s*min-height\s*:\s*@enable-color-preference\s*\)/';
+        $regEx2 = '/\(\s*min-height\s*:\s*@enable-color-preference\s*\)/';
+        $enableAuto = [];
+        if (isset($formData['theme']) && $formData['theme'] !== StyleSheet::DEFAULT_THEME) {
+            $file = file_get_contents(StyleSheet::getThemeFile($formData['theme']));
+            $hasMode = (bool) preg_match_all($regEx, $file, $matches);
+            $onlyAuto = $hasMode ? null : preg_match($regEx2, $file);
+            $enableAuto = ($hasMode || $onlyAuto) ? ['auto'] : [];
+        }
```

Comment:
Uh oh, I don't think it needs to be that complicated. :smile: 

You only need to check for the light mode query: `@media\s*\(\s*min-height\s*:\s*@prefer-light-color-scheme\s*\)\s*,\s*\(\s*prefers-color-scheme\s*:\s*light\s*\)\s*and\s*\(\s*min-height\s*:\s*@enable-color-preference\s*\)`

If it matches, this means that all three modes are available. If it doesn't match, check for any mode query: `@media.*prefers-color-scheme`

If this matches, only the `Auto` mode is available.

Reply:
Oh and please make the regex patterns constants in class `StyleSheet`.